### PR TITLE
Change composer.json to use vcs repositories

### PR DESCRIPTION
**Pull request by kenjis**
Now we use "repoistories - type -package". But it has a few limitations.
See Note on https://getcomposer.org/doc/05-repositories.md#package-2

If you merge this PR, we will have no limitations.

But now we use `--prefer-dist` and `--prefer-source` with special meanings, "source" means 1.8/develop (git) and "dist" means 1.7/master (zip file). If you merge this PR, you can't use the trick. And after merging, you have to change "version" properly before releases.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,136 +5,24 @@
...
         "php": ">=5.3.3",
         "composer/installers": "~1.0",
-        "fuel/docs": "1.7.2",
-        "fuel/core": "1.7.2",
-        "fuel/auth": "1.7.2",
-        "fuel/email": "1.7.2",
-        "fuel/oil": "1.7.2",
-        "fuel/orm": "1.7.2",
-        "fuel/parser": "1.7.2",
+        "fuel/docs": "dev-1.8/develop",
```
Comment (Art4) [inline composer.json]: Can `fuel/docs` be moved to require-dev? Who need docs in production?

**Comment by kenjis** [comment]
@Art4 Thank you for your comment. I did.

**Comment by WanWizard** [comment]
I think it is more logical and in line with how composer works to use the required version. On the other hand, you now always get a git clone, while previously it would download the zip?

**Comment by kenjis** [comment]
Because the required version has `develop` stability. `dev-` means develop.
So the default is `git clone`.

You can install via Zip download if you use `--prefer-dist` like this:

```
$ git clone git@github.com:fuel/fuel.git --depth=1 --branch=1.8/develop
$ cd fuel
$ vi composer.json  # use this PR composer.json
$ composer install --prefer-dist

$ find . -name .git
./.git
```

If you create a tag like `v1.7.3` and use it in `composer.json`, probably composer's default is Zip download.

**Pull request merged**